### PR TITLE
ext_proc: Stop timers during ExternalProcessing::Filter::onDestroy

### PR DESCRIPTION
**Pull request by ristewar**
Signed-off-by: Rick Stewart <ristewart@google.com>

Commit Message: If a timer fires after onDestroy is called, the ext_proc filter may crash while trying to access invalid encoder or decoder filter callbacks. The documentation for StreamFilterBase::onDestroy() says, "Filters must not invoke either encoder or decoder filter callbacks after having onDestroy() invoked." envoy-security@googlegroups.com said it's OK to upstream this since ext_proc is marked alpha.

Additional Description:
Risk Level: Low
Testing: Unit tests added
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by mpwarres — APPROVED** [review]

**Comment by mpwarres** [comment]
You might also consider adding a test case to [ext_proc_integration_test.cc](https://github.com/envoyproxy/envoy/blob/main/test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc) similar to  [ExtProcIntegrationTest.ResponseMessageTimeout](https://github.com/envoyproxy/envoy/blob/0a2aca4729a853d107ec7e2b89a557ba45aa16a8/test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc#L1203), but with the added element that the downstream disconnects before time is advanced.

File: source/extensions/filters/http/ext_proc/processor_state.cc [hunk 1]
```diff
@@ -43,6 +42,12 @@ void ProcessorState::onFinishProcessorCall(Grpc::Status::GrpcStatus call_status,
   callback_state_ = next_state;
 }
 
+void ProcessorState::stopTimers() {
```
[the remark this answers is not in the document]
Comment (ristewar) [reply source/extensions/filters/http/ext_proc/processor_state.cc]: Done, renamed to `stopMessageTimer`.

**Comment by ristewar** [comment]
> You might also consider adding a test case to [ext_proc_integration_test.cc](https://github.com/envoyproxy/envoy/blob/main/test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc) similar to [ExtProcIntegrationTest.ResponseMessageTimeout](https://github.com/envoyproxy/envoy/blob/0a2aca4729a853d107ec7e2b89a557ba45aa16a8/test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc#L1203), but with the added element that the downstream disconnects before time is advanced.

Done.

**Review by mpwarres — DISMISSED** [review]

**Review by snowp — APPROVED** [review]
LGTM, thanks! CI seems to be in a bad state, can you take a look?

**Comment by ristewar** [comment]
> LGTM, thanks! CI seems to be in a bad state, can you take a look?

Thanks! I fixed the tests.

**Review by snowp — APPROVED** [review]
Thanks!

**Pull request merged**